src/vector/ops/set.rs

```diff
@@ -32,6 +32,22 @@ impl Geometry {
         }
         Some(unsafe { Geometry::with_c_geometry(ogr_geom, true) })
     }
+
+    pub fn union(&self, other: &Self) -> Option<Self> {
+        if !self.has_gdal_ptr() {
+            return None;
+        }
+        if !other.has_gdal_ptr() {
+            return None;
+        }
+        unsafe {
+            let ogr_geom = gdal_sys::OGR_G_Union(self.c_geometry(), other.c_geometry());
+            if ogr_geom.is_null() {
+                return None;
```

Comment:
I haven't tested, but it might be possible to return a proper error here?

But then I'm not sure what we should do in the `!has_gdal_ptr()` case...

Reply:
According to gdal's document, when an error occurs, the return value is null, so here we return None, which seems to be consistent with gdal.

Reply:
Yeah, but in most cases GDAL will set an error code and message like [here](https://github.com/OSGeo/gdal/blob/master/ogr/ogrgeometry.cpp#L4531), even without documenting it explicitly.

But indeed, this doesn't seem to be done in a consistent fashion.